internal/workflow/batch.go

```diff
@@ -85,3 +114,92 @@ func (w *BatchWorkflow) updateBatch(ctx temporalsdk_workflow.Context, state *bat
 
 	return nil
 }
+
+func (w *BatchWorkflow) startSIPWorkflow(
+	ctx temporalsdk_workflow.Context,
+	state *batchWorkflowState,
+	index int,
+	key string,
+	sourceID uuid.UUID,
+) error {
+	// Generate SIP UUID using SideEffect to ensure determinism.
+	var sipUUID uuid.UUID
+	genUUID := temporalsdk_workflow.SideEffect(ctx, func(ctx temporalsdk_workflow.Context) any {
+		return uuid.Must(uuid.NewRandomFromReader(w.rng))
+	})
+	if err := genUUID.Get(&sipUUID); err != nil {
+		return fmt.Errorf("generate SIP UUID: %v", err)
+	}
+
+	// Create SIP.
+	sip := datatypes.SIP{
+		UUID:     sipUUID,
+		Name:     key,
+		Status:   enums.SIPStatusQueued,
+		Batch:    &state.batch,
+		Uploader: state.batch.Uploader,
+	}
+	activityOpts := withLocalActivityOpts(ctx)
+	err := temporalsdk_workflow.ExecuteLocalActivity(
+		activityOpts,
+		createSIPLocalActivity,
+		w.ingestsvc,
+		&createSIPLocalActivityParams{SIP: sip},
+	).Get(activityOpts, nil)
+	if err != nil {
+		return fmt.Errorf("create SIP: %v", err)
+	}
+
+	// Start processing workflow for the SIP, keeping track of the workflow future and execution.
+	var we temporalsdk_workflow.Execution
+	processingCtx := temporalsdk_workflow.WithChildOptions(ctx, temporalsdk_workflow.ChildWorkflowOptions{
+		Namespace:         w.cfg.Temporal.Namespace,
+		TaskQueue:         w.cfg.Temporal.TaskQueue,
+		WorkflowID:        fmt.Sprintf("%s-%s", ingest.ProcessingWorkflowName, sipUUID.String()),
+		ParentClosePolicy: temporalapi_enums.PARENT_CLOSE_POLICY_TERMINATE,
```

Comment:
If your intention is to run the child SIP workflows asynchronously, then the docs say you need to use an "abandon" `ParentClosePolicy`: https://docs.temporal.io/develop/go/child-workflows#async-child-workflows